TPL/force-app/main/default/classes/HealthcareCostTriggerHandler.cls

```diff
@@ -1,165 +1,50 @@
 public with sharing class HealthcareCostTriggerHandler {
     public static Boolean isFirstTime = true;
     public static void updateRollup(List<Healthcare_Cost__c> hccList){
-        System.debug(hccList);
         set<Id> caseIds = new set<Id>();
         for(Healthcare_Cost__c hcc : hccList){
             caseIds.add(hcc.Case2__c);
         }
-        System.debug(caseIds);
-        List<Healthcare_Cost__c> hccSumList = [select id,Total_Cost_Override__c,case2__c,RecordTypeName__c,Cost__c,Cost_Include__c,Cost_Review__c from Healthcare_Cost__c where 
-                                               case2__c IN:caseIds];
-        System.debug(hccSumList);
+        List<AggregateResult> groupedResults = [select SUM(Total_Cost_Override__c)tot,case2__c,RecordType.DeveloperName,sum(Cost__c)costSum from Healthcare_Cost__c where Case2__c IN:caseIds and ((Cost_Include__c = true and Cost_Review__c = true) or (RecordType.DeveloperName IN ('Continuing_Care','Future_Care')))    GROUP BY case2__c,RecordType.DeveloperName] ;
         List<Case> caseUpdateList = new List<Case>();
-        Decimal totalCostOverride =0;
```

Comment:
This section was restored from a full looping calculation back to the original aggregrate query method.